client/src/js/services/journalGridService.js

```diff
@@ -0,0 +1,338 @@
+angular.module('bhima.services')
+.service('JournalGridService', ['JournalDataviewService', 'JournalColumnsService', 'JournalManagerService', 'uuid', 'SessionService', 'messenger', 'JournalValidationService', 'liberror', 'JournalDataLoaderService', 'util', 'connect', '$q', '$translate', 'precision', '$window', 
```

Comment:
Could you use the style we've been using so far? Like this:
```js
angular.module('some.module')
.service('SomeService', SomeService);

SomeService.$inject = [ /* dependencies */];

function SomeService(/* aliases for the dependencies */ ) {
  var service = this;
  / * code */
}
```

Using the standard name `var service = this;` will help me review this more effectively, since I'll always know what `this` is.